Source/Rewards/Challenges/ChainedTogether.cs

```diff
@@ -1,11 +1,273 @@
-﻿using WCSharp.Api;
+﻿using System;
+using System.Collections.Generic;
 using static WCSharp.Api.Common;
+using WCSharp.Api;
+using System.Linq;
 
 public static class ChainedTogether
 {
+    private static Dictionary<string, lightning> KittyLightnings = new Dictionary<string, lightning>();
+    private static List<List<Kitty>> kittyGroups;
+    private static float timerInterval = 0.1f;
+    private static Random rng = Globals.RANDOM_GEN;
 
+    private static readonly Dictionary<DifficultyLevel, (int good, int far, int breakPoint)> ranges = new()
+    {
+        { DifficultyLevel.Normal,     (400, 600, 800) },
+        { DifficultyLevel.Hard,       (450, 650, 850) },
+        { DifficultyLevel.Impossible, (500, 700, 900) }
+    };
```

Comment:
Don't forget nightmare! i doubt anyones doing it on nightmare tho
```suggestion
    private static readonly Dictionary<DifficultyLevel, (int good, int far, int breakPoint)> ranges = new()
    {
        { DifficultyLevel.Normal,     (400, 600, 800) },
        { DifficultyLevel.Hard,       (450, 650, 850) },
        { DifficultyLevel.Impossible, (500, 700, 900) },
        { DifficultyLevel.Nightmare,  (550, 750, 950) }
    };
    ```